docs/draft/rest_api.md

```diff
@@ -0,0 +1,111 @@
+# First Draft for the REST API
+All calls and requests return data as JSON.
+## Data that needs to be provided by the API
+- current status of the transformer
+  ```
+  GET status/
+  ```
+  **returns:**
+  ```json
+  {
+    "state": "error|transformation|idle",
+    "current_transformations": [1,2,3]
+  }
+  ```
+  *state*: current status of the transformer
+  *current_transformations:* ids of the active or queued transformations
+- the result of the transformation: file/archive
+  ```
+  GET transformation/{id}
+  ```
+  **on success returns:**
+  ```json
+  {
+    "id": 1,
+    "result": "file"
+  }
+  ```
+- status of the current transformation
+  ```
+  GET transformation/{id}/status
+  ```
+  **on success returns:**
+  ```json
+  {
+    "id": 1,
+    "status": "inqueue|started|stopped|failed",
+    "progress":  "1-100"
+  }
+  ```
+  *status*: returns the current status
+  *progress*: returns the progress as percentage from 1-100%
+- provide log files for a specific transformation
+  ```
+  GET transformation/{id}/log
```

Comment:
Maybe we should discuss the behaviour of this. I think putting the whole log into one Sting is not the right solution. I think about the following changes/additons:

- replace the one string with a Array of strings. Each element represents one line in the log.
- because those logs can get very long we should implement some mechanism that limits the linecount by default. For example: If you call the request (with no extra parameter) you should receive the most recent 50 lines of the log. If you want more or everything a parameter has to be added.